src/BSON/Javascript.c

```diff
@@ -84,9 +165,17 @@ ZEND_BEGIN_ARG_INFO_EX(ai_Javascript___construct, 0, 0, 1)
 	ZEND_ARG_INFO(0, scope)
 ZEND_END_ARG_INFO();
 
+ZEND_BEGIN_ARG_INFO_EX(ai_Javascript___set_state, 0, 0, 1)
+	ZEND_ARG_ARRAY_INFO(0, properties, 0)
+ZEND_END_ARG_INFO();
+
+ZEND_BEGIN_ARG_INFO_EX(ai_Javascript_void, 0, 0, 0)
+ZEND_END_ARG_INFO();
+
 static zend_function_entry php_phongo_javascript_me[] = {
 	PHP_ME(Javascript, __construct, ai_Javascript___construct, ZEND_ACC_PUBLIC|ZEND_ACC_FINAL)
-	PHP_ME(Manager, __wakeUp, NULL, ZEND_ACC_PUBLIC)
+	PHP_ME(Javascript, __set_state, ai_Javascript___set_state, ZEND_ACC_PUBLIC|ZEND_ACC_STATIC)
+	PHP_ME(Javascript, __wakeup, ai_Javascript_void, ZEND_ACC_PUBLIC)
```

Comment:
Javascript and Binary are both missing a `__toString()` implementation ([PHPC-729](https://jira.mongodb.org/browse/PHPC-729)).

Javascript also has no getters for code and scope ([PHPC-730](https://jira.mongodb.org/browse/PHPC-730)).